File: ios/runanywhere-sdks-latest/sdk/runanywhere-react-native/packages/core/src/Public/Extensions/RunAnywhere+Logging.ts
/**
 * RunAnywhere+Logging.ts
 *
 * Logging extension for RunAnywhere SDK.
 * Matches iOS: RunAnywhere+Logging.swift
 */

import type { LogLevel } from '../../Foundation/Logging/Models/LogLevel';
import type {
  LogEventCallback,
  LogDestination,
} from '../../Foundation/Logging';

// ============================================================================
// Logging Extension
// ============================================================================

/**
 * Set SDK log level
 * Matches iOS: static func setLogLevel(_ level: LogLevel)
 */
export function setLogLevel(level: LogLevel): void {
  const { LoggingManager } = require('../../Foundation/Logging');
  LoggingManager.shared.setLogLevel(level);
}

/**
 * Subscribe to all SDK log events
 * Matches iOS pattern of exposing log events publicly.
 */
export function onLog(callback: LogEventCallback): () => void {
  const { LoggingManager } = require('../../Foundation/Logging');
  return LoggingManager.shared.onLog(callback);
}

/**
 * Add a custom log destination
 * Matches iOS: static func addLogDestination(_ destination: LogDestination)
 */
export function addLogDestination(destination: LogDestination): void {
  const { LoggingManager } = require('../../Foundation/Logging');
  LoggingManager.shared.addDestination(destination);
}

/**
 * Remove a log destination by identifier
 */
export function removeLogDestination(identifier: string): void {
  const { LoggingManager } = require('../../Foundation/Logging');
  LoggingManager.shared.removeDestination(identifier);
}

File: ios/runanywhere-sdks-latest/sdk/runanywhere-swift/Sources/RunAnywhere/Public/Extensions/RunAnywhere+Logging.swift
//
//  RunAnywhere+Logging.swift
//  RunAnywhere SDK
//
//  Extension for configuring logging
//

import Foundation

extension RunAnywhere {

    // MARK: - Logging Configuration

    /// Configure logging with a predefined configuration
    /// - Parameter config: The logging configuration to apply
    public static func configureLogging(_ config: LoggingConfiguration) {
        Logging.shared.configure(config)
    }

    /// Enable or disable local console logging
    /// - Parameter enabled: Whether to enable local logging
    public static func setLocalLoggingEnabled(_ enabled: Bool) {
        Logging.shared.setLocalLoggingEnabled(enabled)
    }

    /// Set minimum log level for SDK logging
    /// - Parameter level: Minimum log level to capture
    public static func setLogLevel(_ level: LogLevel) {
        Logging.shared.setMinLogLevel(level)
    }

    /// Enable or disable Sentry error tracking
    /// - Parameter enabled: Whether to enable Sentry logging
    public static func setSentryLoggingEnabled(_ enabled: Bool) {
        Logging.shared.setSentryLoggingEnabled(enabled)
    }

    /// Add a custom log destination
    /// - Parameter destination: The destination to add
    public static func addLogDestination(_ destination: LogDestination) {
        Logging.shared.addDestination(destination)
    }

    // MARK: - Debugging Helpers

    /// Enable verbose debugging mode
    /// - Parameter enabled: Whether to enable verbose mode
    public static func setDebugMode(_ enabled: Bool) {
        setLogLevel(enabled ? .debug : .info)
        setLocalLoggingEnabled(enabled)
    }

    /// Force flush all pending logs to destinations
    public static func flushLogs() {
        Logging.shared.flush()
    }
}
